lib/ReactViews/Workbench/Controls/SummaryConcept/SummaryConcept.jsx

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+import flattenNested from '../../../../Core/flattenNested';
+import ActiveConcept from './ActiveConcept';
+import OpenInactiveConcept from './OpenInactiveConcept';
+import Icon from '../../../Icon.jsx';
+import ObserveModelMixin from '../../../ObserveModelMixin';
+import React from 'react';
+import Styles from './summary-concept.scss';
+
+const ADD_FIRST_TEXT = 'Add a condition';
+const ADD_MORE_TEXT = 'Add new condition';
+
+/*
+ * SummaryConcept displays all the active and open nodes under a given
+ * SummaryConcept.
+ * It has two "modes":
+ *   When summaryConcept.allowMultiple is false, it can be substituted directly for
+ *   a regular displayVariableConcept (or array of concepts). Each child can have
+ *   zero, one or more items selected or not selected (depending on their allowMultiple setting).
+ *   When summaryConcept.allowMultiple is true, it treats the child concepts as
+ *   "conditions", any number of which can be added or removed.
+ *
+ * Parents containing 1 or more active nodes are shown via <./ActiveConcept>.
+ *    (They may be open or closed, and ActiveConcept handles the difference.)
+ * Open nodes not containing any active nodes are shown via <./OpenInactiveConcept>.
+ *    (This is typically the case when a user has pressed the AddButton but yet to
+ *    activate any leaf nodes.)
+ *    (If summaryConcept.allowMultiple is false, you cannot 'cancel' or 'go back' on this.)
+ * If summaryConcept.allowMultiple is true, then an <./AddButton> is also shown,
+ *    which simply opens the root concept, at which point OpenInactiveConcept takes over.
+ *
+ * This design would need revision to handle concepts whose direct children are a mix of
+ * both leaf nodes and parent nodes.
+ *
+ * This component cheekily uses the active concepts' isActive flag -
+ * which is by rights the boolean true - to instead store an integer (>0) giving
```

Comment:
you're right, that's a trick I was using in the cotf2 branch for something else, which doesn't appear here. I'll remove it from this PR.